sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/trees/TreeNodeSuite.scala

```diff
@@ -810,4 +810,15 @@ class TreeNodeSuite extends SparkFunSuite with SQLHelper {
         fail("TreeNode.nodeName should not throw malformed class name error")
     }
   }
+
+  object Spark37800 {
+    case class TestNode(set: Set[String]) extends LogicalPlan with LeafNode {
```

Comment:
I think we can just define here without outer `Spark37800`.

Reply:
In a large codebase, it's important to keep developer mental load to a minimum. One way to do this is to make the intent of definitions very clear.

If we remove the scoping of `TestNode` there may be ambiguity as to its purpose. Is it meant to be used just in the one test or does it have a broader purpose? If more than one test uses this class, it makes more sense for it to have scope and/or name independent of a specific test. Alas, `FunSuite` in ScalaTest doesn't create automatic test scopes the way, say, `WordSpec` does.

Anyway, if you prefer `TestNode` being unscoped, I'll be happy to remove `Spark37800`. In that case, would you be OK with renaming `TestNode` to `Spark37800Node`?

Reply:
Yeah, I know some projects use this way (adding issue ticket as the name of class/objects) but it's not quite usual in Apache Spark codebase up to my best knowledge.

We can even place this class as a nested class within `test("SPARK-37800:... ")` - I think that would address your points.

Reply:
> We can even place this class as a nested class within `test("SPARK-37800:... ")`

Good idea.